fitnessllm_dataplatform/utils/logging_utils.py

```
@@ -41,0 +1,131 @@
+        **kwargs
+    ) -> None:
+        """Log warning level message with structured data."""
+        log_data = self._format_log("WARNING", message, data_source, user_id, **kwargs)
+        self.logger.warning(json.dumps(log_data))
+
+    def debug(
+        self,
+        message: str,
+        data_source: Optional[str] = None,
+        user_id: Optional[str] = None,
+        **kwargs
```

Ruff (0.8.2)

Missing type annotation for `**kwargs`

(ANN003)